server/src/test/java/org/elasticsearch/ingest/IngestDocumentTests.java

```diff
@@ -1029,6 +1029,7 @@ public void testCopyConstructorWithZonedDateTime() {
 
         assertThat(copy.getSourceAndMetadata().get("beforeClockChange"), equalTo(original.getSourceAndMetadata().get("beforeClockChange")));
         assertThat(copy.getSourceAndMetadata().get("afterClockChange"), equalTo(original.getSourceAndMetadata().get("afterClockChange")));
+        assertThat(copy.executedPipelines, equalTo(original.executedPipelines));
```

Comment:
If you compare the two `getPipelineStack` results then you don't need the increased visibility, and I'd rather you didn't need the increased visibility. You could also add a copying getter that's just for visibility, but if you squint that's what `getPipelineStack` is already.

Reply:
Stupid mistake here though -- executedPipelines is always empty in this test. I'm adding a new test for this, and making executedPipelines private again.